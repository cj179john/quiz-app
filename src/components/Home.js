import React, {useEffect} from 'react';
import { Link } from 'react-router-dom';
import {createStyles, makeStyles} from '@material-ui/core/styles';
import {connect} from 'react-redux';
import { getActivities } from './Activities/actions';
import {map, values} from 'ramda';
import { List, ListItem, ListItemText, Divider } from '@material-ui/core';

const useStyles = makeStyles(theme =>
  createStyles({
    questionBox: {
      [theme.breakpoints.up('md')]: {
        padding: theme.spacing(8),
        maxWidth: 345,
      }
    },
    question: {
      border: 1
    },
    answer: {
      padding: theme.spacing(8)
    }
  })
);

function Home(props) {
  const {activities, dispatch} = props;

  const classes = useStyles();
  const links = (id) => ({
    1: `/activities/${id}/questions`,
    2: `/activities/${id}/rounds`
  });

  useEffect(() => {
    dispatch(getActivities());
  }, [dispatch]);

  return (
    <List component="nav" className={classes.root} aria-label="result-list">
      <ListItem>
        <ListItemText primary="Quiz" />
      </ListItem>
      <ListItem>
        <ListItemText primary='Results' />
      </ListItem>
      <Divider light />
      {
        map(activity => (
          <ListItem button
            variant="contained"
            color="primary"
            component={Link}
            to={links(activity.id)[activity.id]}
          >
            {activity.name}
          </ListItem>
        ), values(activities))
      }
    </List>
  );
}

const mapStateToProps = (state) => ({
  activities: state.activities.byId
});

const mapDispatchToProps = (dispatch) => ({dispatch});

export default connect(mapStateToProps, mapDispatchToProps)(Home);