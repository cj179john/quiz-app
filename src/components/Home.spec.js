import { render } from '@testing-library/react';
import { createLocation, createMemoryHistory } from 'history';
import React from 'react';
import {Provider} from 'react-redux';
import { MemoryRouter } from 'react-router';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {Home} from './Home';

describe('Home Component', () => {
  let dispatch;
  let store;
  let props;
  let component;

  afterEach(() => jest.clearAllMocks());

  beforeEach(() => {
    const createMockStore = configureMockStore([thunk]);
    const initialState = {
      activities: { byCode: {} }
    };

    store = createMockStore(initialState);

    const history = createMemoryHistory({initialEntries: []});
    const path = '/';

    const mockMatch = {
      isExact: false,
      params: {},
      path,
      url: path
    };

    const location = createLocation(mockMatch.url, {}, 'testKey');

    dispatch = jest.fn();

    props = {
      dispatch,
      history,
      location,
      match: mockMatch
    };
    component = render(
      <MemoryRouter initialEntries={[{pathname: `/`, key: 'testKey'}]}>
        <Provider store={store}>
          <Home {...props}/>
        </Provider>
      </MemoryRouter>
    );
  });

  it('should match the snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});