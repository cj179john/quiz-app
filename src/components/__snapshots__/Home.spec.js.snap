// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="result-list"
        class="MuiList-root MuiList-padding"
      >
        <li
          class="MuiListItem-root makeStyles-header-1 MuiListItem-gutters"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4"
          >
             Quiz 
          </h4>
        </li>
        <hr
          class="MuiDivider-root MuiDivider-light"
        />
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="result-list"
      class="MuiList-root MuiList-padding"
    >
      <li
        class="MuiListItem-root makeStyles-header-1 MuiListItem-gutters"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
           Quiz 
        </h4>
      </li>
      <hr
        class="MuiDivider-root MuiDivider-light"
      />
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
